airflow/providers/apache/spark/hooks/spark_submit.py

```diff
@@ -404,11 +404,16 @@ def submit(self, application="", **kwargs):
         # Check spark-submit return code. In Kubernetes mode, also check the value
         # of exit code in the log, as it may differ.
         if returncode or (self._is_kubernetes and self._spark_exit_code != 0):
-            raise AirflowException(
-                "Cannot execute: {}. Error code is: {}.".format(
-                    self._mask_cmd(spark_submit_cmd), returncode
+            # double check by spark driver pod status (blocking function)
+            spark_driver_pod_status = self._start_k8s_pod_status_tracking()
```

Comment:
This is going to fail hard when not in kubenetes mode.

Reply:
Yes, thanks for that, I've split the 'if' conditions to remove the influence when not in k8s mode.